airbyte-integrations/connectors/source-zenloop/source_zenloop/components.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+import math
+from dataclasses import dataclass
+from typing import Any, Iterable, List, Mapping, Optional
+
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.declarative.requesters.paginators.strategies.page_increment import PageIncrement
+from airbyte_cdk.sources.declarative.stream_slicers.datetime_stream_slicer import DatetimeStreamSlicer
+from airbyte_cdk.sources.declarative.stream_slicers.substream_slicer import SubstreamSlicer
+from airbyte_cdk.sources.declarative.types import StreamSlice, StreamState
+
+
+@dataclass
+class ZenloopDatetimeStreamSlicer(DatetimeStreamSlicer):
+    def get_request_params(
+        self,
+        *,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Mapping[str, Any]:
+        params = super().get_request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        date_shortcut = {"date_shortcut": "custom"}
```

Comment:
I don't think we need a custom stream slicer here. It looks like we can use the default `DatetimeStreamSlicer` and add a request param

```
        request_options_provider:
          request_parameters:
            order_type: "desc"
            order_by: "inserted_at"
            date_shortcut: "custom"
```

Reply:
Updated